### ISSUE TITLE
Vector observation mismatch during inference - mlagents 0.15.1

### ISSUE THREAD

**Issue by fedetask** [Unity-Technologies/ml-agents#3736]
**Describe the bug**
Using mlagents `0.15.1`. My agent has a Vector observation of `Space size = 10`, `Stacked vectors = 1`. 
In the code I add 10 float observations at each step. When I train the model I don't get any error, but when I try to run it I get a vector observation mismatch error.

**To Reproduce**
This is my `CollectObservations()` method. 
```
public override void CollectObservations(VectorSensor sensor) {
            Vector3 ball_relative_pos =
                transform.InverseTransformDirection(ball.transform.position - transform.position);
            sensor.AddObservation(ball_relative_pos.x / 200.0f);
            sensor.AddObservation(ball_relative_pos.y / 200.0f);
            sensor.AddObservation(ball_relative_pos.z / 200.0f);

            Vector3 ball_relative_vel =
                transform.InverseTransformDirection(ball_rBody.velocity);
            sensor.AddObservation(ball_relative_vel.x / 50.0f);
            sensor.AddObservation(ball_relative_vel.y / 50.0f);
            sensor.AddObservation(ball_relative_vel.z / 50.0f);
            
            Vector3 velocity_relative = transform.InverseTransformDirection(self_rBody.velocity);
            sensor.AddObservation(velocity_relative.x / 50f);  // Drift speed
            sensor.AddObservation(velocity_relative.z / 50f);

            Vector3 other_goal_relative_pos =
                transform.InverseTransformDirection(other_goal.transform.position - transform.position);
            sensor.AddObservation(other_goal_relative_pos.x / 200.0f);
            sensor.AddObservation(other_goal_relative_pos.z / 200.0f);
        }
```
There is no other call to `sensor.AddObservation()` in the code.

**Console logs / stack traces**
```
mismatch between vector observation size (42) and number of observations written (10)
UnityEngine.Debug:AssertFormat(Boolean, String, Object[])
MLAgents.Inference.VectorObservationGenerator:Generate(TensorProxy, Int32, IEnumerable`1) (at /home/fedetask/Desktop/ml-agents-0.15.1/com.unity.ml-agents/Runtime/Inference/GeneratorImpl.cs:122)
MLAgents.Inference.TensorGenerator:GenerateTensors(IEnumerable`1, Int32, IEnumerable`1) (at /home/fedetask/Desktop/ml-agents-0.15.1/com.unity.ml-agents/Runtime/Inference/TensorGenerator.cs:141)
MLAgents.Inference.ModelRunner:DecideBatch() (at /home/fedetask/Desktop/ml-agents-0.15.1/com.unity.ml-agents/Runtime/Inference/ModelRunner.cs:165)
MLAgents.Policies.BarracudaPolicy:DecideAction() (at /home/fedetask/Desktop/ml-agents-0.15.1/com.unity.ml-agents/Runtime/Policies/BarracudaPolicy.cs:60)
MLAgents.Agent:DecideAction() (at /home/fedetask/Desktop/ml-agents-0.15.1/com.unity.ml-agents/Runtime/Agent.cs:805)
MLAgents.Academy:EnvironmentStep() (at /home/fedetask/Desktop/ml-agents-0.15.1/com.unity.ml-agents/Runtime/Academy.cs:467)
MLAgents.AcademyFixedUpdateStepper:FixedUpdate() (at /home/fedetask/Desktop/ml-agents-0.15.1/com.unity.ml-agents/Runtime/Academy.cs:34)

```

**Screenshots**
My agent editor configuration
![Screenshot from 2020-04-04 12-18-07](https://user-images.githubusercontent.com/32360377/78424507-66b77280-766e-11ea-92e0-75f9461668f6.png)

**Environment (please complete the following information):**
- OS + version:Ubuntu 18.04
- _ML-Agents version_: ML-Agents v0.15.1
- _TensorFlow version_: 2.0.1

**Comment by fedetask** [comment]
The weird thing is that the training works without raising any error. But if I stop the training, set the generated model in the agent and run, then I start getting the exceptions.

**Comment by fedetask** [comment]
It turned out that the models/ folder generated during training was outside Assets/, and when I was selecting the model on the Unity Editor I was actually selecting a different model inside Assets/ (that had the same name).

Sorry

**Comment by manuelott85** [comment]
THANK YOU! It took me hours to finally made the connection:
I also had an old trained model assigned with my agent. After increasing the number of observations I got the error "mismatch between vector observation size (163) and number of observations written (166)". And when I increased the observation size by 3 to match 166, then the number of written ones also increased by 3!
It turned out that the old model could not work with these different circumstances. Of course it doesn't and I never intended it to be able to do that, but I just couldn't link the error message with the old model. Assigning "none" fixed the issue. Thank you for providing this hint.

I guess ML-Agent's Error Feedback could be a little better :)
